### PR TITLE
[e2e] Refactor end-to-end tests to use directory

### PR DESCRIPTION
**Pull request by jaylees14**
Ah. Quite big. You'll be wanting to just filter to `.scala` files 😄 

Writing E2E tests was getting annoying, and we were missing cases where we didn't fully check certain subdirectories etc. So: now we compare a directory in `resources` with the output files by building a map of both and then comparing. This means the test logic never has to change (unless you add a new templefile).

I found an issue with the way `FileMatchers` was working, it has been corrected to `actual shouldBe expected`. 

After this PR I'll go round and make sure all our tests do this, as it makes for some hard diffing 😢

### CODE REVIEW

File: src/e2e/resources/simple-temple-expected/auth-db/init.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE auth (
+  id UUID UNIQUE NOT NULL,
+  email TEXT UNIQUE NOT NULL,
+  password TEXT NOT NULL
+);
```
Comment (c-harding) [inline src/e2e/resources/simple-temple-expected/auth-db/init.sql:5]: File without trailing newline 😢

File: src/e2e/resources/simple-temple-expected/booking-db/init.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+CREATE TABLE booking (
+
+);
```
Comment (c-harding) [inline src/e2e/resources/simple-temple-expected/booking-db/init.sql:3]: Best table! Also, newline

File: src/e2e/resources/simple-temple-expected/grafana/provisioning/dashboards/simple-temple-test-user.json [hunk 3]
```diff
@@ -882,4 +882,4 @@
   "title" : "SimpleTempleTestUser Dashboard",
   "uid" : "simple-temple-test-user",
   "version" : 1
-}
+}
```
Comment (c-harding) [inline src/e2e/resources/simple-temple-expected/grafana/provisioning/dashboards/simple-temple-test-user.json:885]: newline

File: src/e2e/scala/temple/E2ETestUtils.scala [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  /**
+    * Build a map of files present in the provided filepath
+    * @param path The root path to explore
+    * @return A map of files, relative to the provided path
+    */
+  def buildFileMap(path: Path): FileSystem.Files = {
+    var foundFiles: FileSystem.Files = Map.empty
+    Files.walkFileTree(
+      path,
+      new FileVisitor[Path] {
```
Comment (c-harding) [inline src/e2e/scala/temple/E2ETestUtils.scala:22]: such scotlandyard, very Tilo

**Review by c-harding — APPROVED** [review]
Feels like there’s a random mix of trailing newlines and not in the generated code. Not your fault, but GitHub makes it more obvious now

The scala looks beautifully OOP

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀

**Pull request merged**